packages/playground/src/components/project-settings/members-list.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useReducer } from 'react'
+import {
+  Icon,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+  Text,
+  Avatar,
+  AvatarImage,
+  AvatarFallback,
+  Badge,
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuGroup,
+  DropdownMenuItem,
+  DropdownMenuSeparator,
+  DropdownMenuShortcut,
+  DropdownMenuTrigger
+} from '@harnessio/canary'
+import { getInitials } from '../../utils/utils'
+import { FormEditMemberDialog } from './form-member-edit-dialog'
+import { FormDeleteMemberDialog } from './form-member-delete-dialog'
+import { dialogStateReducer, initialDialogState } from './members-reducers/dialog-state-reducers'
+
+interface MembersProps {
+  display_name: string
+  role: string
+  email: string
+  timestamp?: string
+  avatarUrl?: string
+}
+
+interface PageProps {
+  members: MembersProps[]
+}
+
+export const MembersList = ({ members }: PageProps) => {
+  const [dialogState, dispatch] = useReducer(dialogStateReducer, initialDialogState)
+
+  const openDialog = (dialogType: 'edit' | 'delete', member: MembersProps) => {
+    dispatch({ type: 'OPEN_DIALOG', dialogType, member })
+  }
+
+  const closeDialog = (dialogType: 'edit' | 'delete') => {
+    dispatch({ type: 'CLOSE_DIALOG', dialogType })
+  }
+
+  // Delete project handler
+  const handleDelete = () => {
+    dispatch({ type: 'START_DELETING' })
+
+    setTimeout(() => {
```

Comment:
also why are timeouts used here?

Reply:
Looks like this is for simulating user flow in `playground` application. But I've asked @jessieyu777 to move this out so that it could be handled the right way with api integration in `gitness` app.

Reply:
move to the page list so will not affect this component in gitness, please see the new commit in this PR,
cc: @cjlee01 , @vardanbansal-harness